syntax = "proto3";

package masterworker;

// The store service definition.
service MasterWorker {
  // Puts worker to work with either a map task or a reduce task
        rpc DoMap (FileChunk) returns (MapStatus) {}
        rpc DoReduce (FileChunk) returns (ReduceStatus) {}
        rpc CheckStatus (Empty) returns (WorkerStatus) {}
}

// The request message containing the file_shard_name
message FileChunk {
        string name = 1;
	int32 round_no = 2;
	int32 n_int_files = 3;
        repeated string temp_files = 4;
	int32 indx = 5;	
}

// The response message containing the status of map task
message MapStatus {
        bool map_status = 1;
        string worker_id = 2;
        repeated string temp_files = 3;
}

// The response message containing the status of map task
message ReduceStatus {
        bool reduce_status = 1;
        string worker_id = 2;
}

message Empty{

}

message WorkerStatus{
  bool worker_status = 1;
}
